cmd/ebpf/xdp/utils/gtp_utils.h

```diff
@@ -79,27 +99,15 @@ static __always_inline long remove_gtp_header(struct packet_context *ctx) {
         bpf_printk("upf: remove_gtp_header: can't set new eth");
         return -1;
     }
-    __builtin_memcpy(new_eth, eth, sizeof(*new_eth));
 
     data += sizeof(*new_eth);
     if (data + 1 > data_end)
         return -1;
 
-    const __u8 ip_version = (*(const __u8 *)data) >> 4;
-    switch (ip_version) {
-        case 6: {
-            new_eth->h_proto = bpf_htons(ETH_P_IPV6);
-            break;
-        }
-        case 4: {
-            new_eth->h_proto = bpf_htons(ETH_P_IP);
-            break;
-        }
-        default:
-            /* do nothing with non-ip packets */
-            bpf_printk("upf: can't process not an ip packet: %d", ip_version);
-            return -1;
-    }
+    if (set_eth_protocol(eth, data) != 0)
```

Comment:
I'd rather get `h_proto`(aka ethertype) as a return value and then set `new_eth->h_proto` **after** memcpy.